client/homebrew/editor/stringArrayEditor/stringArrayEditor.jsx

```diff
@@ -0,0 +1,138 @@
+const React = require('react');
+const createClass = require('create-react-class');
+const _     = require('lodash');
+
+const StringArrayEditor = createClass({
+	displayName     : 'StringArrayEditor',
+	getDefaultProps : function() {
+		return {
+			label         : '',
+			values        : [],
+			valuePatterns : null,
+			placeholder   : '',
+			unique        : false,
+			cannotEdit    : [],
+			onChange      : ()=>{}
+		};
+	},
+
+	getInitialState : function() {
+		return {
+			valueContext : !!this.props.values ? this.props.values.map((value)=>({
+				value,
+				editing : false
+			})) : [],
+			temporaryValue : '',
+			updateValue    : ''
+		};
+	},
+
+	componentDidUpdate : function(prevProps) {
+		if(!_.eq(this.props, prevProps)) {
```

Comment:
I think by definition `componentDidUpdate` is triggered to run whenever `props` is updated, so `this.props != prevProps` will always be true.

Reply:
It also triggers when the state is updated, and since the contents of this `if` are running `setState` it runs until the call stack is exceeded if the `if` statement is removed.

Reply:
Ah, right right. In that case can we have it just check whether `this.props.values` has changed?